oap-server/server-receiver-plugin/skywalking-pprof-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/pprof/provider/handler/PprofServiceHandler.java

```diff
@@ -0,0 +1,130 @@
+import org.apache.skywalking.oap.server.core.storage.StorageModule;
+import org.apache.skywalking.oap.server.core.cache.PprofTaskCache;
+import org.apache.skywalking.oap.server.core.profiling.pprof.storage.PprofTaskLogRecord;
+import org.apache.skywalking.oap.server.core.analysis.worker.RecordStreamProcessor;
+import org.apache.skywalking.oap.server.core.analysis.TimeBucket;
+import java.util.concurrent.TimeUnit;
+
+//import org.apache.skywalking.oap.server.core.storage.StorageModule;
+//import org.apache.skywalking.oap.server.core.storage.profiling.pprof.IPprofTaskQueryDAO;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+import org.apache.skywalking.oap.server.library.server.grpc.GRPCHandler;
+import org.apache.skywalking.oap.server.network.trace.component.command.PprofTaskCommand;
+import org.apache.skywalking.oap.server.receiver.pprof.provider.handler.stream.PprofByteBufCollectionObserver;
+import org.apache.skywalking.oap.server.receiver.pprof.provider.handler.stream.PprofCollectionMetaData;
+import org.apache.skywalking.oap.server.receiver.pprof.provider.handler.stream.PprofFileCollectionObserver;
+
+@Slf4j
+public class PprofServiceHandler extends PprofTaskGrpc.PprofTaskImplBase implements GRPCHandler {
+
+    private final IPprofTaskQueryDAO taskDAO;
+    private final SourceReceiver sourceReceiver;
+    private final CommandService commandService;
+    private final PprofTaskCache taskCache;
+    private final int pprofMaxSize;
+    private final boolean memoryParserEnabled;
+
+    public PprofServiceHandler(ModuleManager moduleManager, int pprofMaxSize, boolean memoryParserEnabled) {
+        this.taskDAO = moduleManager.find(StorageModule.NAME).provider().getService(IPprofTaskQueryDAO.class);
+        this.commandService = moduleManager.find(CoreModule.NAME).provider().getService(CommandService.class);
+        this.sourceReceiver = moduleManager.find(CoreModule.NAME).provider().getService(SourceReceiver.class);
+        this.taskCache = moduleManager.find(CoreModule.NAME).provider().getService(PprofTaskCache.class);
+        this.pprofMaxSize = pprofMaxSize;
+        this.memoryParserEnabled = memoryParserEnabled;
+    }
+    
+    @Override
+    public StreamObserver<PprofData> collect(StreamObserver<PprofCollectionResponse> responseObserver) {
+        return memoryParserEnabled ?
+                new PprofByteBufCollectionObserver(taskDAO, responseObserver, sourceReceiver, pprofMaxSize)
+                : new PprofFileCollectionObserver(taskDAO, responseObserver, sourceReceiver, pprofMaxSize);
+    }
+
+    @Override
+    public void getPprofTaskCommands(PprofTaskCommandQuery request, StreamObserver<Commands> responseObserver) {
+        String serviceId = IDManager.ServiceID.buildId(request.getService(), true);
+        String serviceInstanceId = IDManager.ServiceInstanceID.buildId(serviceId, request.getServiceInstance());
+        PprofTask task = taskCache.getPprofTask(serviceId);
```

Comment:
Since you have https://github.com/apache/skywalking-data-collect-protocol/blob/master/pprof/Pprof.proto#L76 parameter, please use this to query the newer tasks.